src/components/AddPaymentMethodMenu.js

```diff
@@ -42,36 +38,31 @@ const defaultProps = {
 };
 
 const AddPaymentMethodMenu = props => (
-    <Popover
+    <PopoverMenu
         isVisible={props.isVisible}
         onClose={props.onClose}
         anchorPosition={props.anchorPosition}
-    >
-        <View style={StyleUtils.getPaymentMethodMenuWidth(props.isSmallScreenSize)}>
-            <MenuItem
-                title={props.translate('common.bankAccount')}
-                icon={Expensicons.Bank}
-                onPress={() => props.onItemSelected(CONST.PAYMENT_METHODS.BANK_ACCOUNT)}
-                wrapperStyle={styles.pr15}
-            />
-            {Permissions.canUseWallet(props.betas) && (
-                <MenuItem
-                    title={props.translate('common.debitCard')}
-                    icon={Expensicons.CreditCard}
-                    onPress={() => props.onItemSelected(CONST.PAYMENT_METHODS.DEBIT_CARD)}
-                    wrapperStyle={styles.pr15}
-                />
-            )}
-            {props.shouldShowPaypal && !props.payPalMeUsername && (
-                <MenuItem
-                    title={props.translate('common.payPalMe')}
-                    icon={Expensicons.PayPal}
-                    onPress={() => props.onItemSelected(CONST.PAYMENT_METHODS.PAYPAL)}
-                    wrapperStyle={styles.pr15}
-                />
-            )}
-        </View>
-    </Popover>
+        onItemSelected={() => props.onClose()}
+        menuItems={[
+            {
+                text: props.translate('common.bankAccount'),
+                icon: Expensicons.Bank,
+                onSelected: () => props.onItemSelected(CONST.PAYMENT_METHODS.BANK_ACCOUNT),
+            },
+            ...(Permissions.canUseWallet(props.betas) ? [{
+                text: props.translate('common.debitCard'),
+                icon: Expensicons.CreditCard,
+                onSelected: () => props.onItemSelected(CONST.PAYMENT_METHODS.DEBIT_CARD),
+            },
+            ] : []),
```

Comment:
@thesahindia hmm, I prefer that we conditionally push items into the array. Creating a new function is fine. Thoughts?

Reply:
@rushatgabhane, just followed the same approach as we have in `SidebarScreen.js` but I don't mind using a function tho

Reply:
Yeah, it's no big deal. Let's go with function then

Reply:
On second thought, I find both ways _not elegant_ 🤷‍♂️
So let's just leave this as it is